### ISSUE TITLE
[Mapping] Retrieve Positions for Tokens Just in Time

### ISSUE THREAD

**Issue by temanbrcom** [eclipse/che-che4z-lsp-for-cobol#340]
## Description ##
As a SE, I would like to retrieve the original positions for tokens in CobolVisitior just in time, so I can remove the extra work of collecting unnecessary position mapping elements.

## Acceptance Criteria ##
- Syntax and semantic positions remain the same
- No changes in syntax analysis
- The original positions retrieving applied only for tokens that need it

## Definition of Done ##
- The story is implemented according to the description.
- Unit testing done (Preferably by TDD approach, coverage should be at least 70%).
- Code review done by at least one other team member.
- Manual testing is done.
- Documentation is added and reviewed (if needed).
- The pull request is approved.
- The story is merged into the development branch.
- Tests are performed on the development branch.

## Technical details ##
Blocked by #339
